src/Jira/Api/Client/CurlClient.php

```diff
@@ -52,10 +52,6 @@ public function __construct()
      */
     public function sendRequest($method, $url, $data = array(), $endpoint, AuthenticationInterface $credential, $isFile = false, $debug = false)
     {
-        if (!($credential instanceof Basic)) {
```

Comment:
Instead of removing use approach taken in `PHPClient`:

```php
if (!($credential instanceof Basic) && !($credential instanceof Anonymous)) {
    throw new \Exception(sprintf("PHPClient does not support %s authentication.", get_class($credential)));
}
```